lib/options.c

```diff
@@ -40,371 +40,249 @@
 
 #include "log.h"
 #include "options.h"
-
 #include "tpm2_util.h"
 
 /*
- * A structure to map a string name to an element in the TCTI_TYPE
- * enumeration.
- */
-typedef struct {
-    char       *name;
-    TCTI_TYPE   type;
-} tcti_map_entry_t;
-/*
- * A table of tcti_map_entry_t structures. This is how we map a string
- * provided on the command line to the enumeration.
+ * Default TCTI: this is a bit awkward since we allow users to enable /
+ * disable TCTIs using ./configure --with/--without magic.
+ * As simply put as possible:
+ * if the tabrmd TCTI is enabled, it's the default.
+ * else if the socket TCTI is enabled it's the default.
+ * else if the device TCTI is enabled it's the default.
+ * We do this to preserve the current default / expected behavior (use of
+ * the socket TCTI).
  */
-tcti_map_entry_t tcti_map_table[] = {
-#ifdef HAVE_TCTI_DEV
-    {
-        .name = "device",
-        .type = DEVICE_TCTI,
-    },
-#endif
-#ifdef HAVE_TCTI_SOCK
-    {
-        .name = "socket",
-        .type = SOCKET_TCTI,
-    },
-#endif
 #ifdef HAVE_TCTI_TABRMD
-    {
-        .name = "tabrmd",
-        .type = TABRMD_TCTI,
-    },
+  #define TCTI_DEFAULT_STR  "abrmd"
+#elif HAVE_TCTI_SOCK
+  #define TCTI_DEFAULT_STR  "socket"
+#elif  HAVE_TCTI_DEV
+  #define TCTI_DEFAULT_STR  "device"
 #endif
-    {
-        .name = "unknown",
-        .type = UNKNOWN_TCTI,
```

Comment:
You are removing this unknown TCTI, but I guess it's OK since I never understood what was the point of it. Better to have proper check and report an error if user attempts to use a non-valid TCTI.